MySaver/Views/StoresPage.xaml.cs

```diff
@@ -7,15 +7,42 @@ public partial class StoresPage : ContentPage
     private StoresViewModel viewModel;
     public StoresPage(StoresViewModel viewModel)
     {
-        InitializeComponent();
+        InitializeComponent();       
         BindingContext = viewModel;
         this.viewModel = viewModel;
     }
-
+    
     protected async override void OnAppearing()
     {
         base.OnAppearing();
         await viewModel.UpdateStoresAsync();
     }
+    
+    public void ShopsFilterByName(object sender, TextChangedEventArgs e)
+    {
+       
+        var searchTerm = e.NewTextValue;
+
+        if (string.IsNullOrWhiteSpace(searchTerm))
+        {
+            searchTerm = string.Empty;
+        }
+
+        searchTerm = searchTerm.ToLowerInvariant();
 
+        var filteredItems = viewModel.Stores.Where(Store => Store.Name.ToLowerInvariant().Contains(searchTerm)).ToList();
+
+
+        foreach(var Store in viewModel.Stores)
```

Comment:
This foreach seems redundant, as the variable "filteredItems" is already the result you need, so my suggestion would be using:
`viewModel.MyItems.Clear();
        foreach(var Item in filteredItems)
        {
            viewModel.MyItems.Add(Item);
        }`
instead as this should work a bit faster (no need to do if checks on every item)